server/src/main/java/org/opentosca/toscana/plugins/lifecycle/TransformationLifecycle.java

```diff
@@ -0,0 +1,45 @@
+package org.opentosca.toscana.plugins.lifecycle;
+
+/**
+ The basic interface to implement the TransformationLifecycle
+ */
+public interface TransformationLifecycle {
+    /**
+     Checks if the Model (of the context that has been given during construction) is valid for this plugin
+     That means the model gets checked for invalid properties such as unsupported Operating Systems or Architectures
+
+     @return True if the vailidation succeeded, false otherwise.
+     Returning false will cause the transformation to fail
+     */
+    boolean checkModel();
+
+    /**
+     Has to implement the prepare lifecycle operations
+     <p>
+     The Prepare lifecycle phase is described as follows:
+     in this step the plugin processes the graph to transform it in the following step.
+     For example the kubernetes plugin needs to split the graph into container and pods.
+     <p>
+     @throws Exception If a exception gets out of the method the transformation will fail
+     */
+    void prepare() throws Exception;
```

Comment:
Just realized that `throws Exception` is a really bad idea here:
When implementing these methods, it's easy to forget error handling of any checked exception (compiler does not complain..)
I suggest to ditch `throws Exception`. In the executing code one has to make sure to catch all unchecked exceptions.